src/main/java/Parser.java

```diff
@@ -0,0 +1,186 @@
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser class takes care of the parsing of full commands that the user typed.
+ */
+public class Parser {
+    public Parser() {
+
+    }
+
+    /**
+     * This function handles the parsing of command recognised by the bot.
+     * @param fullCommand Command that the user typed.
+     * @return Command from enum.
+     * @throws DukeException Exception is thrown when the user typed other commands that the bot do not understand.
+     */
+    public Command parseCommand(String fullCommand) throws DukeException {
+        String[] splitBySpace;
+        splitBySpace = fullCommand.split(" ");
+
+        try {
+            if (fullCommand.equals("")) {
+                throw new NullPointerException("");
+            }
+
+            switch (splitBySpace[0]) {
+            case "list":
+                return Command.LIST;
+            case "done":
+                return Command.DONE;
+            case "delete":
+                return Command.DELETE;
+            case "todo":
+                return Command.TODO;
+            case "event":
+                return Command.EVENT;
+            case "deadline":
+                return Command.DEADLINE;
+            case "find":
+                return Command.FIND;
+            default:
+                throw new DukeException("Don't understand");
+            }
+        } catch (DukeException e) {
+            System.out.println(e);
+        } catch (NullPointerException e) {
+            System.out.println("Please enter a command.");
+        }
+
+        return null;
+    }
+
+    /**
+     * This function will parse the description of todo command.
+     * @param fullCommand Command that the user typed.
+     * @return The description of todo command.
+     */
+    public String parseDescription(String fullCommand) {
```

Comment:
Maybe you could try combining the parsing for different commands together. Combine that with having a command class and subclasses will make your code neater and with fewer repeats.